electrum/payment_identifier.py

```diff
@@ -0,0 +1,719 @@
+
+    def get_fields_for_GUI(self):
+        recipient = None
+        amount = None
+        description = None
+        validated = None
+        comment = None
+
+        if self.emaillike and self.openalias_data:
+            address = self.openalias_data.get('address')
+            name = self.openalias_data.get('name')
+            recipient = self.emaillike + ' <' + address + '>'
+            validated = self.openalias_data.get('validated')
+            if not validated:
+                self.warning = _('WARNING: the alias "{}" could not be validated via an additional '
+                                 'security check, DNSSEC, and thus may not be correct.').format(self.emaillike)
+
+        elif self.bolt11 and self.wallet.has_lightning():
+            recipient, amount, description = self._get_bolt11_fields(self.bolt11)
+
+        elif self.lnurl and self.lnurl_data:
+            domain = urllib.parse.urlparse(self.lnurl).netloc
+            recipient = f"{self.lnurl_data.metadata_plaintext} <{domain}>"
+            amount = self.lnurl_data.min_sendable_sat if self.lnurl_data.min_sendable_sat else None
+            description = self.lnurl_data.metadata_plaintext
+            if self.lnurl_data.comment_allowed:
+                comment = self.lnurl_data.comment_allowed
+
+        elif self.bip70 and self.bip70_data:
+            pr = self.bip70_data
+            if pr.error:
+                self.error = pr.error
+                return
+            recipient = pr.get_requestor()
+            amount = pr.get_amount()
+            description = pr.get_memo()
+            validated = not pr.has_expired()
+            # note: allow saving bip70 reqs, as we save them anyway when paying them
+            #for btn in [self.send_button, self.clear_button, self.save_button]:
+            #    btn.setEnabled(True)
+            # signal to set fee
+            #self.amount_e.textEdited.emit("")
+
+        elif self.spk:
+            pass
+
+        elif self.multiline_outputs:
+            pass
+
+        elif self.bip21:
+            recipient = self.bip21.get('address')
+            amount = self.bip21.get('amount')
+            label = self.bip21.get('label')
+            description = self.bip21.get('message')
+            # use label as description (not BIP21 compliant)
+            if label and not description:
+                description = label
+
+        return recipient, amount, description, comment, validated
+
+    def _get_bolt11_fields(self, bolt11_invoice):
+        """Parse ln invoice, and prepare the send tab for it."""
+        lnaddr = lndecode(bolt11_invoice) #
+        pubkey = lnaddr.pubkey.serialize().hex()
+        for k, v in lnaddr.tags:
+            if k == 'd':
+                description = v
+                break
+        else:
+            description = ''
+        amount = lnaddr.get_amount_sat()
+        return pubkey, amount, description
+
+    async def resolve_openalias(self) -> Optional[dict]:
```

Comment:
try copy-pasting into the qt send tab payto_edit on *testnet*:
`donate@getmonero.org`
On master, it shows an error popup saying invalid address; on this PR it raises -- in both cases only when you click "pay..." -- maybe it should already complain before, as soon as the DNS host is resolved, not sure.

Reply:
Pay.. button is now disabled if the resolved openalias contains an invalid address